modin/pandas/dataframe.py

```diff
@@ -465,6 +465,11 @@ def groupby(
                 level, by = by, None
             elif level is None:
                 by = self.__getitem__(by)._query_compiler
+                if not by.equals_partitioning(self._query_compiler):
+                    # this updates '_modin_frame' inplace
+                    self._query_compiler._modin_frame._filter_empties(
+                        compute_metadata=False
+                    )
```

Comment:
we should not disclose that much information about partitioning at the front-end. More over, the 'empty-partition' problem is relevant not only for groupby but for all of the operations where we broadcast partitions of one frame to another. 

Can we move this logic to the low-level dataframe and perform it before any broadcasting operation?

Reply:
Makes sense, I moved it to a lower level.

> More over, the 'empty-partition' problem is relevant not only for groupby but for all of the operations where we broadcast partitions of one frame to another.

The problem is that `modin_frame.broadcast_apply` does `_copartition` unlike `partition_manager.broadcast_apply` that is used in `groupby_reduce`. I added an assert just in case.